### PR TITLE
Fix #2150: Remove Null Pointer Exception from Speaker

### LINKED ISSUE #2150: App crashes on clicking Speakers from Nav Drawer
**Actual Behaviour**

App Crashes on Selecting Speakers from Navigation Drawer

**Expected Behaviour**

It should not crash

**LogCat for the issue**

FATAL EXCEPTION: main
                                                                    Process: org.fossasia.openevent, PID: 891
                                                                    java.lang.NullPointerException: Attempt to invoke virtual method 'void org.fossasia.openevent.dbutils.FilterableRealmLiveData.filter(io.reactivex.functions.Predicate)' on a null object reference
                                                                        at org.fossasia.openevent.viewmodels.SpeakersListFragmentViewModel.loadFilteredSpeakers(SpeakersListFragmentViewModel.java:44)
                                                                        at org.fossasia.openevent.viewmodels.SpeakersListFragmentViewModel.getSpeakers(SpeakersListFragmentViewModel.java:35)
                                                                        at org.fossasia.openevent.fragments.SpeakersListFragment.loadData(SpeakersListFragment.java:117)
                                                                        at org.fossasia.openevent.fragments.SpeakersListFragment.onCreateView(SpeakersListFragment.java:87)
                                                                        at android.support.v4.app.Fragment.performCreateView(Fragment.java:2261)
                                                                        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1419)
                                                                        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1750)
                                                                        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1819)
                                                                        at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:797)
                                                                        at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2590)
                                                                        at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2377)
                                                                        at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2332)
                                                                        at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2239)
                                                                        at android.support.v4.app.FragmentManagerImpl$1.run(FragmentManager.java:700)
                                                                        at android.os.Handler.handleCallback(Handler.java:751)
                                                                        at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                        at android.os.Looper.loop(Looper.java:154)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:6776)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1518)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1408)

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by S2606**
Fixes #2150

### CODE REVIEW

**Comment by S2606** [comment]
@srv-twry Please review and thank you for helping me.
@iamareebjamal Please review

**Review by dr0pdb — APPROVED** [review]

**Comment by dr0pdb** [comment]
@S2606 Thanks for finding the issue. It was a really bad mistake from my side.

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java [hunk 1]
```diff
@@ -27,8 +27,10 @@ public SpeakersListFragmentViewModel() {
 
     public LiveData<List<Speaker>> getSpeakers(int sortType, String searchText) {
         setSearchText(searchText);
-        if (sortType != speakersListSortType) {
-            filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSpeakers(sortOrderSpeaker()));
+        if (sortType != speakersListSortType || filterableRealmLiveData == null) {
+            if(filterableRealmLiveData == null) {
```
Comment (dr0pdb) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java]: Please remove this null check. I thought it needs to be created only if the filterableRealmLiveData is null but it is also using ```sortOrderSpeaker()``` inside the ```realmRepo.getSpeakers``` call.

File: android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java [hunk 2]
```diff
@@ -27,8 +27,10 @@ public SpeakersListFragmentViewModel() {
...
+                filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSpeakers(sortOrderSpeaker()));
+            }
```
Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java]:
```Java
if(filterableRealmLiveData == null) {
    filterableRealmLiveData = RealmDataRepository.asFilterableLiveData(realmRepo.getSpeakers(sortOrderSpeaker()));
}
````

This line can be moved out of the if statement I think?

Comment (iamareebjamal) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java]: And handled above?

Comment (dr0pdb) [reply android/app/src/main/java/org/fossasia/openevent/viewmodels/SpeakersListFragmentViewModel.java]: @iamareebjamal Yes.I suggested him to add it but I missed the fact that filterableRealmLiveData will also change when the sortOrder is different.

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**